src/p2p/net_node.inl

```diff
@@ -1556,6 +1556,19 @@ namespace nodetool
   }
   //-----------------------------------------------------------------------------------
   template<class t_payload_net_handler>
+  void node_server<t_payload_net_handler>::get_peerlist(std::vector<std::vector<peerlist_entry>>& gray, std::vector<std::vector<peerlist_entry>>& white)
```

Comment:
Propose to return merged gray and white lists, `zone.second.m_peerlist.get_peerlist` appends new entries to the lists, it doesn't clear them.

```c++
void node_server<t_payload_net_handler>::get_peerlist(std::vector<peerlist_entry>& gray, std::vector<peerlist_entry>& white)
{
    for (auto &zone: m_network_zones)
    {
      zone.second.m_peerlist.get_peerlist(gray, white);
    }
}
```

Reply:
If it doesn't break any caller, sure.

Reply:
`node_server<t_payload_net_handler>::get_peerlist` is a new function introduced in this PR, changing it doesn't break any caller.

Implementing it the way i propose will omit a fair amount of changed lines in `core_rpc_server::on_get_peer_list` as well.

Reply:
OK, I see what you mean now.